Xamarin.Forms.Platform.iOS/Renderers/NavigationRenderer.cs

```diff
@@ -1213,6 +1216,12 @@ void UpdateToolbarItems()
 				List<UIBarButtonItem> secondaries = null;
 				foreach (var item in _tracker.ToolbarItems)
 				{
+					item.PropertyChanged -= OnToolbarItemPropertyChanged;
+					item.PropertyChanged += OnToolbarItemPropertyChanged;
```

Comment:
Why do we need to unsubscribe and subscribe to the same thing right after each other?

Also, shouldn't the `item.IsVisible` check be between here? No need to hook the event to that when it's not visible, right?

Reply:
@jfversluis 
Thank you for reviewing.

That is because it avoids duplicate subscriptions.

Because every time a toolbaritem's IsVisible property is changed, invoked  UpdateToolbarItems method, it should remove an old subscription and add a new subscription.

Even if the item is not visible, it needs to event hook for changing from invisible to visible.
If the item.IsVisible check is between here, it can't restore visible state.

Thanks.

Reply:
@jfversluis 
Did you solve your question?
I would be grateful if you could resolve conversation for my answer.